### ISSUE TITLE
How to calculate landmark loss?

### ISSUE THREAD

**Issue by xinwen-cs** [microsoft/Deep3DFaceReconstruction#36]
Hi,
    I am inplementing landmark loss, I am confused about eqaution (3) in 4.1.2 of your paper. There are two questions.
    <1> Is N always equal to 68, even if there are self-occlusion ?
    <2> Is q_n  and q'_n include z-component when calculating equation (3) ? (I know that you use [8] to detect 3D landmark to get (x,y,z) tuples).

Thanks!

**Comment by YuDeng** [comment]
Hi, we use 68 landmarks for loss computation regardless of the self-occlusion. Landmarks used for loss computation are 2d projections so z-components are excluded. Thanks.

**Comment by Moon-W** [comment]
> Hi,
> I am inplementing landmark loss, I am confused about eqaution (3) in 4.1.2 of your paper. There are two questions.
> <1> Is N always equal to 68, even if there are self-occlusion ?
> <2> Is q_n and q'_n include z-component when calculating equation (3) ? (I know that you use [8] to detect 3D landmark to get (x,y,z) tuples).
> 
> Thanks!

Hi, I'm a rookie about this, could you send me a copy of your training codes? 
My email:ChrisandPaul3@163.com
Thanks a lot !!!

**Comment by xinwen-cs** [comment]
Hi,
    Sorry for bothering again. I want to know what the **nose points** exactly mean in 4.1.2 of your paper? Is it including both nose bridge and nose tip(9 landmarks in total)? And what does **inner mouth** exactly mean? Could you please tell me the indices of landmark you set to 20 in the picture below?

Thanks

![dlib-landmark-mean](https://user-images.githubusercontent.com/25365618/69845911-46853a80-12ad-11ea-8956-84bcd2350696.png)

**Comment by hangon666** [comment]
@xinwen-cs 
Hello, may I ask a question ?
When I tried to reimplenment the loss function, I don't konw how to calculate the landmark loss in the training step, could you share your landmark loss function here or send to my email? Thanks a lot!
My email: cherry_2lin@126.com
